### ISSUE TITLE
The highlight color in comments shows incorrectly after typing @category

### ISSUE THREAD

**Issue by yuyully** [redhat-developer/vscode-java#2884]
##### Environment
- Operating System: Windows
- JDK version: 17.0.3
- Visual Studio Code version: 1.74.3
- Java extension version: 1.14.0

##### Steps To Reproduce
1.  Add a multi-line comments `/* ... */`
2. Type `@apiNote` and `@category`

##### Current Result
![image](https://user-images.githubusercontent.com/31497677/212835154-d0c4cc53-b055-45fb-8552-73274885d6a2.png)

##### Expected Result
![image](https://user-images.githubusercontent.com/31497677/212835238-b27c4343-1558-472a-b14b-3580291d3aff.png)

**Comment by rgrunber** [comment]
@testforstephen I just had a look at https://github.com/microsoft/vscode/blob/main/extensions/java/syntaxes/java.tmLanguage.json#L3-L5 . However, https://github.com/atom/language-java/ is read-only (since end of 2022). I guess it hasn't been an issue since the last change coming from there was from the [start of 2021](https://github.com/atom/language-java/commits/master/grammars/java.cson).

How would we contribute changes to the grammar if needed ? Could we do it directly in the VS Code repo ?

**Comment by 0dinD** [comment]
The TextMate grammar for Java has always been a bit lacking. For the most part, the problems have been hidden away behind the semantic token-based syntax highlighting implemented in JDT LS, which uses the Java AST generated by JDT Core.

But as you can see when using the [Scope Inspector](https://code.visualstudio.com/api/language-extensions/syntax-highlight-guide#scope-inspector), the semantic tokens for the Javadoc tags "break" when adding the `@category` tag. It looks like this is a bug in JDT Core, where the AST breaks if you add the `@category` tag. Interestingly enough, it seems to work fine if you add a word after the tag, such as `@category foo`, the AST is only broken if you have an `@category` tag without a missing word.

The bug in JDT Core can also be reproduced in the Eclipse IDE, for example using the [AST View](https://eclipse.dev/jdt/ui/astview/index.php), if you look at the AST of the Javadoc comment before and after adding the `@category` tag.

---

In the end, I guess it doesn't really matter (to the end user) where we fix the issue (either in the TextMate grammar, to fix the TextMate tokens, or in JDT Core, to fix the semantic tokens). The bug in JDT Core could possibly affect other features that use the AST though.

**Comment by rgrunber** [comment]
Yes, this is definitely an upstream issue. It seems similar in function with https://bugs.eclipse.org/bugs/show_bug.cgi?id=125676 . The `@category` tag seems to read beyond the end of line, treating the next token as an argument to itself.

**Comment by fbricon** [comment]
No longer an issue.

<img width="240" height="131" alt="Image" src="https://github.com/user-attachments/assets/574678d7-616e-444a-9240-43c57d089e21" />

**Comment by fbricon** [comment]
Actually still an issue if the category is empty

<img width="200" height="129" alt="Image" src="https://github.com/user-attachments/assets/ebe60fb0-04ec-40c0-8490-aa225a9d5c6a" />

meh
